x/incentives/keeper/distribute_test.go

```diff
@@ -876,3 +887,109 @@ func (s *KeeperTestSuite) TestDistributeConcentratedLiquidity() {
+	currentEpoch := s.App.EpochsKeeper.GetEpochInfo(s.Ctx, incentivesParams)
+
+	// 1. Initialize variables
+	poolIds := []uint64{1, 2, 3}
+	startTime := s.Ctx.BlockTime()
+	// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+	gauges := s.SetupCLPoolAndGauge(uint64(len(poolIds)), currentEpoch.Duration, sdk.NewCoins(), startTime)
+
+	//Test1: distribute 2 coins using CL gauge
+	gaugeRewards := sdk.NewCoins(sdk.NewCoin("uusdc", sdk.NewInt(10_000)), sdk.NewCoin("uosmo", sdk.NewInt(10_000)))
+	expectedCoinsToDistribute := sdk.NewCoins(sdk.NewCoin("uusdc", sdk.NewInt(30_000)), sdk.NewCoin("uosmo", sdk.NewInt(30_000))) // 3 pool, 10k each token per pool = 30k each token
+	s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(10_000_000_000)),
+		sdk.NewCoin("uusdc", sdk.NewInt(10_000_000_000)),
+		sdk.NewCoin("uatom", sdk.NewInt(10_000_000_000)),
+	))
+
+	s.FunctionalConcentratedDistributeGaugeHelper(s.Ctx, poolIds, gauges, startTime, gaugeRewards, expectedCoinsToDistribute)
+
+	//Test2: distribute 3 coins using CL gauge
+	poolIds1 := []uint64{4, 5, 6}
+	gauges1 := s.SetupCLPoolAndGauge(uint64(len(poolIds1)), currentEpoch.Duration, sdk.NewCoins(), startTime)
+	gaugeRewards1 := sdk.NewCoins(sdk.NewCoin("uusdc", sdk.NewInt(10_000)), sdk.NewCoin("uosmo", sdk.NewInt(10_000)), sdk.NewCoin("uatom", sdk.NewInt(10_000)))
+	expectedCoinsToDistribute1 := sdk.NewCoins(sdk.NewCoin("uusdc", sdk.NewInt(30_000)), sdk.NewCoin("uosmo", sdk.NewInt(30_000)), sdk.NewCoin("uatom", sdk.NewInt(30_000))) // 3 pool, 10k each token per pool = 30k each token
+
+	s.FunctionalConcentratedDistributeGaugeHelper(s.Ctx, poolIds1, gauges1, startTime, gaugeRewards1, expectedCoinsToDistribute1)
+}
+
+func (s *KeeperTestSuite) FunctionalConcentratedDistributeGaugeHelper(ctx sdk.Context, poolIds []uint64, gauges_original []types.Gauge, startTime time.Time, gaugeRewards sdk.Coins, expectedCoinsToDistribute sdk.Coins) {
+	// 3. update the gauge by adding rewards (should happen at end of each epoch)
+	for _, gauge := range gauges_original {
+		// adds token from a specific owner to the moduleAddress, which updated the gauge coins in the process
+		err := s.App.IncentivesKeeper.AddToGaugeRewards(ctx, s.TestAccs[0], gaugeRewards, gauge.Id)
+		s.Require().NoError(err)
+	}
+
+	// make sure the module recieved the funds
+	moduleAddress := s.App.AccountKeeper.GetModuleAddress(types.ModuleName)
+	actualModuleCoins := s.App.BankKeeper.GetAllBalances(ctx, moduleAddress)
+	s.Require().Equal(expectedCoinsToDistribute, actualModuleCoins)
+
+	// 4. get updated gauges and distribute the rewards that was added
+	for _, gauge := range gauges_original {
+		err := s.App.IncentivesKeeper.MoveUpcomingGaugeToActiveGauge(ctx, gauge)
+		s.Require().NoError(err)
+	}
+
+	gauges_afterEpoch := s.App.IncentivesKeeper.GetActiveGauges(ctx)
+	s.Require().Equal(len(gauges_original), len(gauges_afterEpoch))
```

Comment:
what would this mean? Are we trying to verify that all the given gauges were supposed to be active regardless of `MoveUpcomingGaugeToActiveGauge`?